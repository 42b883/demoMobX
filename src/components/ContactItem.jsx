import React, { Component } from 'react';

class ContactItem extends Component {
    
    render() {

        return (
            <div className="post card" key={this.props.id}>
                <div className="card-content">
                <span className="card-title blue-text text-darken-2">{this.props.name}</span>
                <p>username: {this.props.username}</p>
                <p>email: {this.props.email}</p>
                <p>website: {this.props.website}</p>
                <p>phone: {this.props.phone}</p>
                {/* <p>hometown: {this.props.hometown}</p> */}
             </div>
            </div>        
        );
    }
}

export default ContactItem
