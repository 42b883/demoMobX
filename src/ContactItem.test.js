/* eslint-env test */

import React from "react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "mobx-react";
import { observable, action, decorate } from "mobx";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import ContactItem from "./components/ContactItem.jsx";

configure({ adapter: new Adapter() });

it("renders each contact item without crashing", () => {
  const props = {
    ContactsStore: {
      name: "Yauhen",
      phone: "23979247",
      website: "www.tt.co",
      email: "tldrl@mail.us"
    }
  };
  const wrapper = shallow(<ContactItem {...props} />);

  expect(wrapper).toMatchSnapshot();
});
it("check count of <p></p>", () => {
  const props = {
    ContactsStore: {
      name: "Yauhen",
      phone: "23979247",
      website: "www.tt.co",
      email: "tldrl@mail.us",
      home: "Ottawa"
    }
  };
  const wrapper = mount(<ContactItem {...props} />);
  expect(wrapper.find("p").length).toEqual(5);
});
