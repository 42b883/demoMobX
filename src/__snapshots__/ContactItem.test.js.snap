// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders each contact item without crashing 1`] = `
<ContactItem
  ContactsStore={
    Object {
      "email": "tldrl@mail.us",
      "name": "Yauhen",
      "phone": "23979247",
      "website": "www.tt.co",
    }
  }
/>
`;
